src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -15,7 +37,374 @@ class CnfNfdGenerator(NFDGenerator):
+                # Unpack the chart into the tmp folder
+                self._extract_chart(helm_package.path_to_chart)
+
+                # TODO: Validate charts
+
+                # Get schema for each chart (extract mappings and take the schema bits we need from values.schema.json)
+                # + Add that schema to the big schema.
+                self.deployment_parameter_schema[
+                    "properties"
+                ].update(self.get_chart_mapping_schema(helm_package))
+                
+                # Get all image line matches for files in the chart.
+                # Do this here so we don't have to do it multiple times.
+                image_line_matches = self.find_pattern_matches_in_chart(
+                    helm_package, IMAGE_LINE_REGEX
+                )
+                
+                # Generate the NF application configuration for the chart
+                self.nf_application_configurations.append(
+                    self.generate_nf_application_config(
+                        helm_package,
+                        image_line_matches,
+                        self.find_pattern_matches_in_chart(
+                            helm_package, IMAGE_PULL_SECRET_LINE_REGEX
+                        ),
+                    )
+                )
+                # Workout the list of artifacts for the chart and
+                # update the list for the NFD with any unique artifacts.
+                chart_artifacts = self.get_artifact_list(
+                    helm_package, set(image_line_matches)
+                )
+                self.artifacts += [
+                    a for a in chart_artifacts if a not in self.artifacts
+                ]
+
+            self.write_nfd_bicep_file()
+            self.write_schema_to_file()
+            self.write_manifest_bicep_file()
+            self.copy_to_output_folder()
+        # Delete tmp folder
+        shutil.rmtree(self.tmp_folder_name)
+
+    @property
+    def bicep_path(self) -> Optional[str]:
+        """Returns the path to the bicep file for the NFD if it has been created."""
+        if os.path.exists(self._bicep_path):
+            return self._bicep_path
+
+        return None
+
+    def _extract_chart(self, fname: str) -> None:
+        """
+        Extract the chart into the tmp folder.
+
+        :param helm_package: The helm package to extract.
+        :type helm_package: HelmPackageConfig
+        """
+        if fname.endswith("tar.gz") or fname.endswith("tgz"):
+            tar = tarfile.open(fname, "r:gz")
+            tar.extractall(path=self.tmp_folder_name)
+            tar.close()
+        elif fname.endswith("tar"):
+            tar = tarfile.open(fname, "r:")
+            tar.extractall(path=self.tmp_folder_name)
+            tar.close()
+        else:
+            # Throw error here
+            shutil.copytree(fname, self.tmp_folder_name, dirs_exist_ok=True)
+
+    def _create_nfd_folder(self) -> None:
```

Comment:
This function isn't called. It looks useful though.  Whatever we do, be consistent between VNF and CNF

Reply:
It's called in custom.py. Have changed to make vnf do the same